src/couch_replicator_httpc_pool.erl

```diff
@@ -172,6 +174,56 @@ terminate(_Reason, State) ->
     lists:foreach(fun ibrowse_http_client:stop/1, State#state.free),
     lists:foreach(fun ibrowse_http_client:stop/1, State#state.busy).
 
+
+release_worker_int(Worker, State0) ->
+    #state{
+        callers = Callers,
+        busy = Busy,
+        closing = Closing
+    } = State0,
+    State0#state{
+        callers = demonitor_client(Callers, Worker),
+        closing = Closing -- [Worker],
+        busy = Busy -- [Worker]
```

Comment:
All three above fail silently. Since with a queue like this, vicious races are a possibility, one would want to add a logging assertion here e.g. that `callers` is now 1 shorter and `closing`, or `busy`, too.